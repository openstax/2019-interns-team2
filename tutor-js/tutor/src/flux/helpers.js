import _ from 'underscore';
import flux from 'flux-react';

const LOADING = 'loading';
const LOADED  = 'loaded';
const FAILED  = 'failed';
const SAVING  = 'saving';
const DELETING = 'deleting';
const DELETED = 'deleted';

const STATES = {
  LOADING,
  LOADED,
  FAILED,
  SAVING,
  DELETING,
  DELETED,
};

let idCounter = 0;
const CREATE_KEY = () => `_CREATING_${idCounter++}`;

const isNew = id => /_CREATING_/.test(id);

const CrudConfig = function() {
  return {
    _asyncStatus: {},
    _local: {},
    _changed: {},
    _errors: {},
    _reload: {},

    // If the specific type needs to do something else to the object:
    // _loaded : (obj, id) ->
    // _saved : (obj, id) ->
    // _reset : ->

    reset() {
      this._asyncStatus = {};
      this._local = {};
      this._changed = {};
      this._errors = {};
      this._reload = {};

      if (typeof this._reset === 'function') {
        this._reset();
      }
      return this.emitChange();
    },

    dontReload(id) {
      return (this._asyncStatus[id] === LOADED) && this._HACK_DO_NOT_RELOAD;
    },

    FAILED(status, msg, id, ...args) {
      this._asyncStatus[id] = FAILED;
      this._errors[id] = msg;
      if (typeof this._failed === 'function') {
        this._failed({ status, msg }, id, ...Array.from(args));
      }

      if (status !== 0) { // indicates network failure
        delete this._local[id];
        return this.emitChange();
      }
    },

    load(id) {
      return this._load(id);
    },

    // allow aliasing load without triggering api call.
    _load(id) {
      // Add a shortcut for unit testing
      if (this.dontReload(id)) { return; }
      this._reload[id] = false;
      this._asyncStatus[id] = LOADING;
      return this.emitChange();
    },

    loaded(obj, id, ...args) {
      // id = obj.id
      this._asyncStatus[id] = LOADED;

      if (obj) {
        // If the specific type needs to do something else to the object:
        this._local[id] = (typeof this._loaded === 'function' ? this._loaded(obj, id, ...Array.from(args)) : undefined) || obj;
      }

      return this.emitChange();
    },

    save(id, obj) {
      // Note: id could be isNew()
      this._asyncStatus[id] = SAVING;
      return this.emitChange();
    },

    saved(result, id, ...args) {
      // id = result.id
      this._asyncStatus[id] = LOADED; // TODO: Maybe make this SAVED

      // If the specific type needs to do something else to the object:
      const obj = typeof this._saved === 'function' ? this._saved(result, id, ...Array.from(args)) : undefined;
      if (obj) { result = obj; }

      if (result) {
        this._local[id] = result;
        this._local[result.id] = result;
        delete this._changed[result.id];
      } else {
        console.warn('API WARN: Server did not return JSON after saving. Patching locally');
        // Merge all the local changes into the new local object
        this._local[id] = _.extend(this._local[id], this._changed[id]);
      }
      delete this._changed[id];
      delete this._errors[id];
      // If the specific type needs to do something else to the object:
      return this.emitChange();
    },

    create(localId, attributes = {}) {
      if (!isNew(localId)) { throw new Error('BUG: MUST provide a local id'); }
      this._local[localId] = {};
      this._changed[localId] = attributes;
      return this._asyncStatus[localId] = LOADED;
    },

    created(result, localId, ...args) {
      this._local[localId] = result; // HACK: So react component can still manipulate the same object
      this._local[result.id] = result;
      this._asyncStatus[localId] = LOADED;
      this._asyncStatus[result.id] = LOADED;
      this.emitChange();
      return (typeof this._created === 'function' ? this._created(result, localId, ...Array.from(args)) : undefined);
    },

    _change(id, obj) {
      if (this._changed[id] == null) { this._changed[id] = {}; }
      _.extend(this._changed[id], obj);
      return this.emitChange();
    },

    _save(id) {
      return this._asyncStatus[id] = SAVING;
    },

    delete(id) {
      this._asyncStatus[id] = DELETING;
      return (typeof this._delete === 'function' ? this._delete(id) : undefined);
    },

    deleted(result, id, ...args) {
      this._asyncStatus[id] = DELETED;
      delete this._local[id];
      this.emitChange();
      return (typeof this._deleted === 'function' ? this._deleted(result, id, ...Array.from(args)) : undefined);
    },

    clearChanged(id) {
      return delete this._changed[id];
    },

    HACK_DO_NOT_RELOAD(bool) { return this._HACK_DO_NOT_RELOAD = bool; },

    // Keep this here so other exports method have access to it
    _get(id) {
      const val = this._local[id];
      if (!val && (this._asyncStatus[id] !== SAVING)) { return null; }
      // Scores stores an Array unlike most other stores
      if (_.isArray(val)) {
        return val;
      } else {
        return _.extend({}, val, this._changed[id]);
      }
    },

    exports: {
      isUnknown(id) { return !this._asyncStatus[id]; },
      isLoading(id) { return this._asyncStatus[id] === LOADING; },
      isLoaded(id) { return this._asyncStatus[id] === LOADED; },
      isDeleting(id) { return this._asyncStatus[id] === DELETING; },
      isSaving(id) { return this._asyncStatus[id] === SAVING; },
      isFailed(id) { return this._asyncStatus[id] === FAILED; },
      getAsyncStatus(id) { return this._asyncStatus[id]; },
      get(id) { return this._get(id); },
      isChanged(id) { return !_.isEmpty(this._changed[id]); },
      getChanged(id) { return this._changed[id] || {}; },
      freshLocalId() { return CREATE_KEY(); },
      isNew(id) { return isNew(id); },
      reload(id) { return this._reload[id]; },
      getError(id) { return this._errors[id]; },
    },
  };
};

// Helper for creating a simple store for actions
const makeSimpleStore = function(storeConfig) {

  let actionsConfig = _.omit(storeConfig, (value, key) => (typeof value !== 'function') || (key === 'exports'));

  actionsConfig = _.keys(actionsConfig);
  const actions = flux.createActions(actionsConfig);
  storeConfig.actions = _.values(actions);
  const store = flux.createStore(storeConfig);
  return { actions, store };
};


const extendConfig = function(newConfig, origConfig) {
  if (newConfig.exports == null) { newConfig.exports = {}; }
  _.defaults(newConfig, origConfig);
  _.defaults(newConfig.exports, origConfig.exports);
  return newConfig;
};


const makeStandardStore = function(name, storeConfig) {
  extendConfig(storeConfig, new CrudConfig());
  const { actions, store } = makeSimpleStore(storeConfig);
  return { [`${name}Actions`]: actions, [`${name}Store`]: store };
};


export { CrudConfig, makeSimpleStore, extendConfig, makeStandardStore, isNew, STATES };
