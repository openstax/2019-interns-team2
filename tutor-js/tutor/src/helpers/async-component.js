import React from 'react';
import Loadable from 'react-loadable';
import ErrorComponent from '../components/error-monitoring/async-load-error';
import Loading from 'shared/components/loading-animation';

export function asyncComponent(loader, name = '') {
  const loading = ({ error }) => {
    return (
      error ? <ErrorComponent error={error} />
        : <Loading message={`Loading ${name}â€¦`} />
    );
  };
  return Loadable({
    loader, loading,
  });

}

export function loadAsync(resolve, name) {
  // return a function so the router will only evaluate it when it's needed
  // in the future we can insert role dependant logic here
  return () => asyncComponent(resolve, name);
}
