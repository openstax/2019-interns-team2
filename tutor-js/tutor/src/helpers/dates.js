import moment from 'moment';
import 'twix';
import 'moment-timezone';
import { reduce } from 'lodash';

export function getDay(oneMoment) {
  return moment(oneMoment)
    .startOf('day')
    .twix(moment(oneMoment).endOf('day'), { allDay: true });
}

export function getDurationFromMoments(listOfMoments) {
  return reduce(
    listOfMoments,
    (current, next) => current.union(getDay(next)),
    getDay(listOfMoments[0])
  );
}
