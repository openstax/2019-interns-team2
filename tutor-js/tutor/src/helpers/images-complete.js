import { isFunction } from 'lodash';

export default function imagesComplete({
  body = document.body,
  timeoutAfter = 10000, // in ms, 10 seconds
} = {}) {
  return new Promise((resolve, reject) => {
    if (!body || !isFunction(body.querySelectorAll)) {
      resolve([]);
      return;
    }

    const images = Array.from(body.querySelectorAll('img'));
    if (0 === images.length) {
      resolve(images);
      return;
    }
    let complete = 0;
    let pendingTimeout = setTimeout(() => {
      pendingTimeout = null;
      reject(images);
    }, timeoutAfter);
    const markComplete = () => {
      complete += 1;
      if (complete === images.length) {
        if (pendingTimeout) {
          clearInterval(pendingTimeout);
          pendingTimeout = null;
        }
        resolve(images);
      }
    };
    images.forEach((img) => {
      if (img.naturalWidth) {
        markComplete();
      } else {
        img.addEventListener( 'load', markComplete, false );
        img.addEventListener( 'error', markComplete, false );
      }
    });
  });

}
