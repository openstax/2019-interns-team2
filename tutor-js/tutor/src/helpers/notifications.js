import { NotificationActions } from 'shared';
import Router from './router';

export default {
  start(bootstrapData) {
    NotificationActions.startPolling();
    return (() => {
      const result = [];
      const object = bootstrapData.flash || {};
      for (let level in object) {
        const message = object[level];
        result.push(NotificationActions.display({ message, level }));
      }
      return result;
    })();
  },

  buildCallbackHandlers(comp) {
    const { router } = comp.context;
    if (!router) {
      throw new Error('Component\'s context must have router present');
    }
    return {
      missing_student_id: {
        onAdd({ course }) {
          return router.history.push(
            Router.makePathname('changeStudentId', { courseId: course.id })
          );
        },
      },
    };
  },
};
