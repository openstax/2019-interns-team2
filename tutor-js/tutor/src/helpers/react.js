import React from 'react';
import ReactDOM from 'react-dom';
import cn from 'classnames';
import { propTypes as mobxPropTypes } from 'mobx-react';
import PropTypes from 'prop-types';
import invariant from 'invariant';
import styled, { css } from 'styled-components';
import { autobind } from 'core-decorators';
import { idType } from 'shared';

export {
  React, cn, ReactDOM, mobxPropTypes, PropTypes,
  invariant, styled, css, idType, autobind,
};
export { observable, action, computed } from 'mobx';
export { observer, inject, Provider } from 'mobx-react';
