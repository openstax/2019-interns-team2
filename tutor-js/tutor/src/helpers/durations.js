import moment from 'moment';
import _ from 'underscore';
import { TimeStore } from '../flux/time';

export default {
  create(startTime, endTime) {
    return moment(startTime).twix(endTime);
  },

  isPastDue({ due_at }) {
    return moment(TimeStore.getNow()).isAfter(due_at);
  },
};
