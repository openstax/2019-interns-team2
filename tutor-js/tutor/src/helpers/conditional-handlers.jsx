import React from 'react';
import { extend } from 'lodash';
import { asyncComponent } from './async-component';
import { CourseNotFoundWarning } from '../components/course-not-found-warning';
import Courses from '../models/courses-map';
import { OXMatchByRouter } from 'shared';

const StudentDashboard = asyncComponent(
  () => import('../screens/student-dashboard')
);

const TeacherDashboard = asyncComponent(
  () => import('../screens/teacher-dashboard')
);

const getConditionalHandlers = (Router) => {
  const MatchForTutor = OXMatchByRouter(Router, null, 'TutorRouterMatch');

  const renderDashboard = (props) => {
    const { courseId } = props.params;

    extend(props, { courseId });
    const course = Courses.get(courseId);

    if (!course) {
      return <CourseNotFoundWarning />;
    }

    if (!props.match.isExact) {
      return (
        <MatchForTutor {...props} />
      );
    }

    if (course.currentRole.isTeacher) {
      return <TeacherDashboard {...props} />;
    } else {
      return <StudentDashboard {...props} />;
    }
  };

  return {
    dashboard() { return renderDashboard; },
  };
};

export { getConditionalHandlers };
