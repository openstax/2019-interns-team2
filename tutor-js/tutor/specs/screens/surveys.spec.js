import { Factory, EnzymeContext } from '../helpers';
import Survey from '../../src/screens/surveys';


describe('Surveys Screen', () => {

  let props, course, surveyRecord;

  beforeEach(() => {
    course = Factory.course();
    course.studentTaskPlans.fetch = jest.fn();
    course.studentTaskPlans.onLoaded({
      data: { tasks: [],
        research_surveys: [Factory.bot.create('ResearchSurvey')],
      },
    });
    surveyRecord = course.studentTaskPlans.researchSurveys.values()[0];
    props = {
      course,
      params: {
        courseId: course.id,
        surveyId: surveyRecord.id,
      },
    };
  });

  it('renders and matches snapshot', () => {
    expect.snapshot(<Survey {...props} />).toMatchSnapshot();
  });

  it('submits survey when answered', async () => {
    const survey = mount(<Survey {...props} />, EnzymeContext.build());

    survey.find('input[placeholder="Jon Snow"]').first().simulate('blur',
      { target: { value: 'Bob' } }
    );

    survey.find('input[type="radio"]').first().simulate('change',
      { target: { value: 'yes' } }
    );
    surveyRecord.save = jest.fn();
    survey.find('.sv_complete_btn').simulate('click', {});
    expect(surveyRecord.save).toHaveBeenCalled();
    expect(surveyRecord.response).toMatchObject({ name: 'Bob' });
    survey.unmount();
  });

});
