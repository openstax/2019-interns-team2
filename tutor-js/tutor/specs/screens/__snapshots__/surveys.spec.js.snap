// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surveys Screen renders and matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  padding: 10px 40px;
  min-height: 100px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  width: 100%;
  max-width: 1200px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  font-weight: bold;
  font-size: 36px;
  line-height: 45px;
  margin: 0;
}

.c4 {
  font-weight: 200;
  line-height: 1em;
  font-size: 1.6rem;
  margin: 0;
}

<div
  className="course-page research-surveys"
  data-appearance="college_physics"
  data-book-title="College Physics"
  data-is-preview={false}
  data-term="Spring 2017"
  data-title="College Physics"
>
  <header>
    <div
      className="title-wrapper c0"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
        >
          <h1
            className="c3"
          >
            College Physics
          </h1>
          <h3
            className="c4"
          >
            Spring 2017
          </h3>
        </div>
      </div>
    </div>
  </header>
  <div
    className="body-wrapper"
  >
    <div
      className="body"
    >
      <h3>
        Do you like Practical Soft Salad
      </h3>
      <div
        className="sv_main sv_default_css"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="sv_custom_header"
          />
          <div
            className="sv_container"
          >
            <div
              className="sv_body"
              id="sp_100"
            >
              <div
                className="sv_p_root"
              >
                <div
                  className="sv_row"
                >
                  <div
                    className="sv_q sv_qstn"
                    id="sq_100"
                    style={
                      Object {
                        "width": "100.000000%",
                      }
                    }
                  >
                    <div
                      className={null}
                    >
                      <h5
                        className="sv_q_title"
                      >
                        <span
                          className="sv_q_num"
                          style={
                            Object {
                              "position": "static",
                            }
                          }
                        >
                          1
                        </span>
                        <span
                          className="sv_q_num"
                        >
                          .
                           
                        </span>
                        <span
                          style={null}
                        >
                          Please enter your name: *
                        </span>
                      </h5>
                    </div>
                    <div
                      className={null}
                    >
                      <input
                        aria-label="Please enter your name: *"
                        className="sv_q_text_root"
                        disabled={false}
                        id="sq_100i"
                        maxLength={null}
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Jon Snow"
                        size={25}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="sv_row"
                >
                  <div
                    className="sv_q sv_qstn"
                    id="sq_101"
                    style={
                      Object {
                        "width": "100.000000%",
                      }
                    }
                  >
                    <div
                      className={null}
                    >
                      <h5
                        className="sv_q_title"
                      >
                        <span
                          className="sv_q_num"
                          style={
                            Object {
                              "position": "static",
                            }
                          }
                        >
                          2
                        </span>
                        <span
                          className="sv_q_num"
                        >
                          .
                           
                        </span>
                        <span
                          style={null}
                        >
                          Your e-mail:
                        </span>
                      </h5>
                    </div>
                    <div
                      className={null}
                    >
                      <input
                        aria-label="Your e-mail:"
                        className="sv_q_text_root"
                        disabled={false}
                        id="sq_101i"
                        maxLength={null}
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="jon.snow@nightwatch.org"
                        size={25}
                        type="email"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="sv_row"
                >
                  <div
                    className="sv_q sv_qstn"
                    id="sq_102"
                    style={
                      Object {
                        "width": "100.000000%",
                      }
                    }
                  >
                    <div
                      className={null}
                    >
                      <h5
                        className="sv_q_title"
                      >
                        <span
                          className="sv_q_num"
                          style={
                            Object {
                              "position": "static",
                            }
                          }
                        >
                          3
                        </span>
                        <span
                          className="sv_q_num"
                        >
                          .
                           
                        </span>
                        <span
                          style={null}
                        >
                          Do you like to eat bananas? *
                        </span>
                      </h5>
                    </div>
                    <div
                      className={null}
                    >
                      <fieldset
                        className="sv_qcbc"
                      >
                        <legend
                          aria-label="Do you like to eat bananas? *"
                        />
                        <div
                          className="sv_q_radiogroup sv-q-col-1"
                        >
                          <label
                            className="sv_q_radiogroup_label"
                          >
                            <input
                              aria-label="No"
                              checked={false}
                              className="sv_q_radiogroup_control_item"
                              disabled={false}
                              id="sq_102i_0"
                              name="bananas_sq_102"
                              onChange={[Function]}
                              type="radio"
                              value="no"
                            />
                            <span
                              className="circle"
                            />
                            <span
                              className="check"
                            />
                            <span
                              className=""
                            >
                              <span
                                style={
                                  Object {
                                    "display": "inline",
                                    "marginLeft": "3px",
                                    "position": "static",
                                  }
                                }
                              >
                                No
                              </span>
                            </span>
                          </label>
                        </div>
                        <div
                          className="sv_q_radiogroup sv-q-col-1"
                        >
                          <label
                            className="sv_q_radiogroup_label"
                          >
                            <input
                              aria-label="Yes, they're great!"
                              checked={false}
                              className="sv_q_radiogroup_control_item"
                              disabled={false}
                              id="sq_102i_1"
                              name="bananas_sq_102"
                              onChange={[Function]}
                              type="radio"
                              value="yes"
                            />
                            <span
                              className="circle"
                            />
                            <span
                              className="check"
                            />
                            <span
                              className=""
                            >
                              <span
                                style={
                                  Object {
                                    "display": "inline",
                                    "marginLeft": "3px",
                                    "position": "static",
                                  }
                                }
                              >
                                Yes, they're great!
                              </span>
                            </span>
                          </label>
                        </div>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  className="sv_row"
                >
                  <div
                    className="sv_q sv_qstn"
                    id="sq_103"
                    style={
                      Object {
                        "width": "100.000000%",
                      }
                    }
                  >
                    <div
                      className={null}
                    >
                      <h5
                        className="sv_q_title"
                      >
                        <span
                          className="sv_q_num"
                          style={
                            Object {
                              "position": "static",
                            }
                          }
                        >
                          4
                        </span>
                        <span
                          className="sv_q_num"
                        >
                          .
                           
                        </span>
                        <span
                          style={null}
                        >
                          Why?
                        </span>
                      </h5>
                    </div>
                    <div
                      className={null}
                    >
                      <textarea
                        aria-label="Why?"
                        className=""
                        cols={50}
                        id="sq_103i"
                        maxLength={null}
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder=""
                        readOnly={false}
                        rows={4}
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="sv_row"
                >
                  <div
                    className="sv_q sv_qstn"
                    id="sq_104"
                    style={
                      Object {
                        "width": "100.000000%",
                      }
                    }
                  >
                    <div
                      className={null}
                    >
                      <h5
                        className="sv_q_title"
                      >
                        <span
                          className="sv_q_num"
                          style={
                            Object {
                              "position": "static",
                            }
                          }
                        >
                          5
                        </span>
                        <span
                          className="sv_q_num"
                        >
                          .
                           
                        </span>
                        <span
                          style={null}
                        >
                          Check other types of fruit you like
                        </span>
                      </h5>
                    </div>
                    <div
                      className={null}
                    >
                      <fieldset
                        className="sv_qcbc sv_qcbx"
                      >
                        <legend
                          aria-label="Check other types of fruit you like"
                        />
                        <div
                          className="sv_q_checkbox sv-q-col-1"
                        >
                          <label
                            className="sv_q_checkbox_label"
                          >
                            <input
                              aria-label="Apples"
                              checked={false}
                              className="sv_q_checkbox_control_item"
                              disabled={false}
                              id="sq_104i_0"
                              onChange={[Function]}
                              style={
                                Object {
                                  "marginRight": "3px",
                                }
                              }
                              type="checkbox"
                              value="apples"
                            />
                            <span
                              className="checkbox-material"
                            >
                              <span
                                className="check"
                              />
                            </span>
                            <span
                              className="sv_q_checkbox_control_label"
                            >
                              <span
                                style={null}
                              >
                                Apples
                              </span>
                            </span>
                          </label>
                        </div>
                        <div
                          className="sv_q_checkbox sv-q-col-1"
                        >
                          <label
                            className="sv_q_checkbox_label"
                          >
                            <input
                              aria-label="Mango"
                              checked={false}
                              className="sv_q_checkbox_control_item"
                              disabled={false}
                              id="sq_104i_1"
                              onChange={[Function]}
                              style={
                                Object {
                                  "marginRight": "3px",
                                }
                              }
                              type="checkbox"
                              value="mango"
                            />
                            <span
                              className="checkbox-material"
                            >
                              <span
                                className="check"
                              />
                            </span>
                            <span
                              className="sv_q_checkbox_control_label"
                            >
                              <span
                                style={null}
                              >
                                Mango
                              </span>
                            </span>
                          </label>
                        </div>
                        <div
                          className="sv_q_checkbox sv-q-col-1"
                        >
                          <label
                            className="sv_q_checkbox_label"
                          >
                            <input
                              aria-label="STEAK!"
                              checked={false}
                              className="sv_q_checkbox_control_item"
                              disabled={false}
                              id="sq_104i_2"
                              onChange={[Function]}
                              style={
                                Object {
                                  "marginRight": "3px",
                                }
                              }
                              type="checkbox"
                              value="steak"
                            />
                            <span
                              className="checkbox-material"
                            >
                              <span
                                className="check"
                              />
                            </span>
                            <span
                              className="sv_q_checkbox_control_label"
                            >
                              <span
                                style={null}
                              >
                                STEAK!
                              </span>
                            </span>
                          </label>
                        </div>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  className="sv_row"
                >
                  <div
                    className="sv_q sv_qstn"
                    id="sq_105"
                    style={
                      Object {
                        "width": "100.000000%",
                      }
                    }
                  >
                    <div
                      className={null}
                    >
                      <h5
                        className="sv_q_title"
                      >
                        <span
                          className="sv_q_num"
                          style={
                            Object {
                              "position": "static",
                            }
                          }
                        >
                          6
                        </span>
                        <span
                          className="sv_q_num"
                        >
                          .
                           
                        </span>
                        <span
                          style={null}
                        >
                          What else would you like to say?
                        </span>
                      </h5>
                    </div>
                    <div
                      className={null}
                    >
                      <textarea
                        aria-label="What else would you like to say?"
                        className=""
                        cols={50}
                        id="sq_105i"
                        maxLength={null}
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder=""
                        readOnly={false}
                        rows={4}
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sv_nav"
              >
                <input
                  className=" sv_complete_btn"
                  onClick={[Function]}
                  onMouseDown={null}
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  type="button"
                  value="Complete"
                />
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
